### PR TITLE
TR-1358 - Templates list page

### PR DESCRIPTION
**Pull request by rajumsys**
Implements templates list view based on mock. As discussed, using `templatesV2` directory and `templatesv2` new route to namespace all v2 work and minimize cleanup/diff for release. This PR is based on #1032; merging that first would reduce diff. 

### What Changed
 - Created a new directory `templatesV2` to keep new designs. 
- In order to minimize cleanup later, refactored routes to support both templates and templatesV2 routing. Any routes missing in v2, will fallback to v1 (/templates). 

### How To Test
 - Checkout to branch
 - Visit `Templates` from `Campaigns` menu. It should take you to new list page. Rest should work as is. 
 - Note: If you click breadcrumbs (like from template edit page), it'll take to old templates route. Those can be fixed after `next` editing paths and breadcrumb links are updated in editing PRs. 

### To Do
- [x] Displaying separate rows for draft & publish
- [x] Implement delete
- [x] Update description 
- [x] Address review feedback

- [x] Change base branch to `master` before merging

Notes:
- Brian used `/next` suffix for all of his routing. As we discussed this, in a later PR, i'll modify that and bring under `/templatev2`

### CODE REVIEW

File: src/config/campaignNavItems.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { hasUiOption } from 'src/helpers/conditions/account';
+import { FilterNone } from '@sparkpost/matchbox-icons/matchbox-icons';
+
+const campaignNavItems = [
+  {
+    label: 'Templates',
+    to: '/templatesv2'
```
Comment (bkemper) [inline src/config/campaignNavItems.js]: 👍

File: src/pages/templatesV2/constants.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export default {
+  routeNamespace: 'templatesv2'
```
Comment (bkemper) [inline src/pages/templatesV2/constants.js]: Probably better exported as a named export.

```
export const routeNamespace = 'templatesv2';
```

**Comment by rajumsys** [comment]
This PR is ready for review

File: src/pages/templatesV2/components/ListComponents.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  );
+};
+
+export const Status = (rowData) => {
+  const listStatus = rowData.list_status;
+
+  const PublishedIcon = <CheckCircle color='#45D09E'/>;
+
+  if (listStatus === 'published') {
+    return <Tag>{PublishedIcon} Published</Tag>;
```
[the remark this answers is not in the document]
Comment (ewandennis) [reply src/pages/templatesV2/components/ListComponents.js]: The icons and text in the `Tag` could use vertical alignment too. I think we did this on the sending domains edit page too. [Here](https://github.com/SparkPost/2web2ui/blame/master/src%2Fcomponents%2FdomainStatus%2FReadyFor.module.scss).

File: src/pages/templatesV2/components/ListComponents.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        </Link>
+      </p>
+    </>
+  );
+};
+
+export const Status = (rowData) => {
+  const listStatus = rowData.list_status;
+
+  const PublishedIcon = <CheckCircle color='#45D09E'/>;
```
Comment (ewandennis) [inline src/pages/templatesV2/components/ListComponents.js]: Is there a better way to handle this hard-coded colour? Our colours are embedded in matchbox at the moment but I believe thats changing. Maybe we should pull this into a CSS file.

File: src/pages/templatesV2/tests/ListPage.test.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  });
+
+  it('renders Loading', () => {
+    const wrapper = subject({ loading: true });
+    expect(wrapper.find('Loading')).toExist();
+  });
+
+  it('renders delete modal visible', () => {
+    const wrapper = subject();
+    wrapper.setState({ showDeleteModal: true });
```
Comment (ewandennis) [inline src/pages/templatesV2/tests/ListPage.test.js]: I expect we could also use Enzyme's `mount()` and simulated a click to get the modal. Not immediately vital though.

File: src/selectors/templates.js [hunk 6]
```diff
@@ -63,3 +63,23 @@ export const selectPublishedTemplatesBySubaccount = createSelector(
   [selectPublishedTemplates, selectSubaccountId, hasSubaccounts],
   (templates, subaccountId, subaccountsExist) => filterTemplatesBySubaccount({ templates, subaccountId, hasSubaccounts: subaccountsExist })
 );
+
+/**
+ * Prepare templates collection for listing with published and draft as separate item.
+ * @return array
+ */
+export const selectTemplatesForListTable = createSelector(
```
Comment (ewandennis) [inline src/selectors/templates.js]: Can we please add some test coverage for this?

File: src/pages/templatesV2/ListPage.js [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+    },
+    {
+      component: LastUpdated,
+      header: {
+        label: 'Last Updated',
+        sortKey: 'last_update_time'
+      }
+    },
+    {
+      component: Action,
```
[the remark this answers is not in the document]
Comment (rajumsys) [reply src/pages/templatesV2/ListPage.js]: There are other actions are coming (updated mock). We'll have to rename to something like `Actions` for that though. If you prefer to have `Delete` for now, let me know.

File: src/pages/templatesV2/components/ListComponents.js (hunk 4 above)
Comment (rajumsys) [reply src/pages/templatesV2/components/ListComponents.js]: CSS file sounds good.

File: src/selectors/templates.js (hunk 6 above)
Comment (rajumsys) [reply src/selectors/templates.js]: :facepalm:

File: src/pages/templatesV2/components/ListComponents.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (rajumsys) [reply src/pages/templatesV2/components/ListComponents.js]: > The icons and text in the Tag could use vertical alignment too. I think we did this on the sending domains edit page too. Here.

Aren't they in the middle now?

[the remark this answers is not in the document]
Comment (ewandennis) [reply src/pages/templatesV2/components/ListComponents.js]: Screen Shot 2019-06-05 at 12 41 16

File: src/pages/templatesV2/ListPage.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (ewandennis) [reply src/pages/templatesV2/ListPage.js]: Naming aside, we should only allow the delete action when the current user has the `templates/modify` grant. That logic exists in the old templates list page but not this one.

**Comment by ewandennis** [comment]
Optional patch to improve tag icon margin.
```
diff --git a/src/pages/templatesV2/components/ListComponents.js b/src/pages/templatesV2/components/ListComponents.js
index e26fc58dd..d33031bc9 100644
--- a/src/pages/templatesV2/components/ListComponents.js
+++ b/src/pages/templatesV2/components/ListComponents.js
@@ -5,6 +5,7 @@ import { CheckCircle, Delete, Edit } from '@sparkpost/matchbox-icons';
 import { formatDateTime } from 'src/helpers/date';
 import { setSubaccountQuery } from 'src/helpers/subaccounts';
 import styles from './ListComponents.module.scss';
+import classnames from 'classnames';
 
 import { routeNamespace } from '../constants';
 
@@ -28,21 +29,26 @@ export const Status = (rowData) => {
 
   const PublishedIcon = <CheckCircle className={styles.PublishedIconColor}/>;
 
+  const tagClasses = classnames(
+    styles.Status,
+    listStatus === 'published_with_draft' && styles.PublishedWithChanges
+  );
+
   if (listStatus === 'published') {
-    return <Tag className={styles.published}>{PublishedIcon} Published</Tag>;
+    return <Tag className={tagClasses}>{PublishedIcon} Published</Tag>;
   }
 
   if (listStatus === 'published_with_draft') {
     return (
       <Tooltip dark content='Contains unpublished changes'>
-        <Tag className={styles.PublishedWithChanges}>
+        <Tag className={tagClasses}>
           {PublishedIcon} Published
         </Tag>
       </Tooltip>
     );
   }
 
-  return <Tag><Edit/> Draft</Tag>;
+  return <Tag className={tagClasses}><Edit/> Draft</Tag>;
 };
 
 export const DeleteAction = ({ onClick, ...props }) => <Button flat onClick={() => onClick(props)}><Delete
diff --git a/src/pages/templatesV2/components/ListComponents.module.scss b/src/pages/templatesV2/components/ListComponents.module.scss
index db3467511..95b567ec7 100644
--- a/src/pages/templatesV2/components/ListComponents.module.scss
+++ b/src/pages/templatesV2/components/ListComponents.module.scss
@@ -15,13 +15,13 @@
   font-size: rem(14);
 }
 
-.PublishedWithChanges {
-  font-style: italic;
+.Status * {
+  margin-bottom: spacing(smaller);
   white-space: nowrap;
 }
 
-.published {
-  white-space: nowrap;
+.PublishedWithChanges {
+  font-style: italic;
 }
 
 .PublishedIconColor {

```

**Pull request merged**